rocky/rocky/views/ooi_detail.py

```diff
@@ -70,10 +64,38 @@ def handle_page_action(self, action: str) -> bool:
                 boefje = get_katalogus(self.organization.code).get_plugin(boefje_id)
                 ooi = self.get_single_ooi(pk=ooi_id)
                 self.run_boefje_for_oois(boefje, [ooi])
-                return True
 
+                success_message = (
+                    "Your scan is running successfully in the background. \n "
+                    "Results will be added to the object list when they are in. "
+                    "It may take some time, a refresh of the page may be needed to show the results."
+                )
+                messages.add_message(self.request, messages.SUCCESS, success_message)
+                return redirect("task_list", organization_code=self.organization.code)
+
+            if action == PageActions.SUBMIT_ANSWER.value:
+                if not isinstance(self.ooi, Question):
+                    messages.add_message(self.request, messages.ERROR, "Only Question OOIs can be answered.")
+                    return self.get(self.request, status_code=500, *self.args, **self.kwargs)
+
+                schema_answer = self.request.POST.get("schema")
+                parsed_schema_answer = json.loads(schema_answer)
+                validator = Draft202012Validator(json.loads(self.ooi.json_schema))
+
+                if not validator.is_valid(parsed_schema_answer):
+                    for error in validator.iter_errors(parsed_schema_answer):
+                        messages.add_message(self.request, messages.ERROR, error.message)
+
+                    return self.get(self.request, status_code=500, *self.args, **self.kwargs)
+
+                self.bytes_client.upload_raw(schema_answer, {"answer", f"{self.ooi.schema_id}"}, self.ooi.reference)
+                messages.add_message(self.request, messages.SUCCESS, "Question has been answered.")
+                return self.get(self.request, status_code=201, *self.args, **self.kwargs)
```

Comment:
I'm not sure if the `status_code` argument does anything. I haven't seen this in Rocky yet and Django doesn't seem to know it. Did you actually get one of these (non-200) status codes rendered in this view?

Reply:
This functionality was based on previous investigation and work (and feedback) at the time about that we want to return 400 error on bad input as well. It actually works as well! Here's e.g. the 201:

![Schermafbeelding 2023-07-25 om 15 54 10](https://github.com/minvws/nl-kat-coordination/assets/46660228/2cb8d49e-ebbd-40cf-932b-b4f655dead0b)